src/core/passableResult/index.js

```diff
@@ -0,0 +1,261 @@
+// @flow
+
+import { WARN, FAIL } from '../../constants';
+const severities: string[] = [ WARN, FAIL ];
+
+type AsyncObject = null | {
+    [fieldName: string]: {
+        done: boolean,
+        callbacks: Function[]
+    }
+};
+
+const passableResult: Function = (name: string): PassableResult => {
+
+    const completionCallbacks: Function[] = [];
+    let asyncObject: AsyncObject = null;
+
+    /**
+     * Initializes specific field's counters
+     * @param {string} fieldName - The name of the field.
+     */
+    const initFieldCounters: Function = (fieldName: string) => {
```

Comment:
I think it would be cleaner to define this function outside the closure. At the moment, it's unpure and  relies on an `output` object to not only be defined, but be defined in a very specific format. The fact that it's defined after the function itself makes it even more confusing.
Same for most functions here.

Reply:
Do you think that they should accept output/asyncObject/completionCallbacks as an argument?

Reply:
Whatever each function needs in order to operate.

Reply:
Overall, that's a valid point, but I think it goes against the idea of using a closure. The closure provide access to lexically scoped variables such as `output`/`asyncObject`/`completionCallbacks`.

Most of these methods are being called from the outside to drive the lifecycle of the suite forward. Exposing these variables outside just to pass them back inside feels counterintuitive to me, and in some cases is simply not possible as in the case of the consumer-facing `.after()` and `.done()` methods.

For the cases in which it _is_ possible, that's something we need to consider in terms of the design of the library.

Putting this aside for further thinking.